bot/notification.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+from telegram import InputMediaPhoto
+from telegram.error import TelegramError
+
+from bot.bot import application
+from bot.models import TelegramIdAd
+from config import telegram_config
+from mongo import db
+
+from app.models import Ad
+
+
+chat_id = telegram_config.id_antalya_chat
+channel_id = telegram_config.id_antalya_channel
+
+logger = logging.getLogger(__name__)
+
+closed_ares = [area["name"] for area in db.areas.find({"is_closed": True})]
+connection_parameters = dict(connect_timeout=20, read_timeout=20)
+
+
+def check_floor(ad: Ad, parameter: list[str]) -> bool:
+    if "without_last" in parameter and ad.data.floor == str(ad.data.building_floor_count):
+        return False
+    if "without_first" in parameter and ad.data.floor in ["Elevation 1", "Garden-Floor"]:
+        return False
+    if "without_basement" in parameter and ad.data.floor in ["Basement", "Ground Floor", "Raised Ground Floor"]:
+        return False
+    return True
+
+
+def check_rooms(ad: Ad, parameter: list[str]) -> bool:
+    if "studio" in parameter and ad.data.room_count == "Studio Flat (1+0)":
+        return True
+    rooms, *_ = ad.data.room_count.split("+")
+    if "one" in parameter and rooms in ["1", "1.5"]:
+        return True
+    if "two" in parameter and rooms in ["2", "2.5"]:
+        return True
+    if "three" in parameter and rooms in ["3", "3.5+1"]:
+        return True
+    if "four" in parameter and rooms in ["4", "4.5", "5", "5.5", "6", "7", "8", "9", "10", "Over 10"]:
+        return True
+    return False
+
+
+def check_heating(ad: Ad, parameter: list[str]) -> bool:
+    if "gas" in parameter and ad.data.heating_type == "Central Heating Boilers":
+        return True
+    if "electricity" in parameter and ad.data.heating_type in ["Elektrikli Radyatör", "Room Heater"]:
+        return True
+    if "central" in parameter and ad.data.heating_type in ["Central Heating", "Central Heating (Share Meter)"]:
+        return True
+    if "underfloor" in parameter and ad.data.heating_type == "Floor Heating":
+        return True
+    if "ac" in parameter and ad.data.heating_type in ["Air Conditioning", "Fan Coil Unit", "VRV", "Heat Pump"]:
+        return True
```

Comment:
когда появляется куча однотипных IF, то следует задуматься можно ли это упростить
```suggestion
        heat_mapping = {
            "gas": {"Central Heating Boilers"},
            "electricity": {"Elektrikli Radyatör", "Room Heater"},
            "central": {"Central Heating", "Central Heating (Share Meter)"},
            "underfloor": {"Floor Heating"},
            "ac": {"Air Conditioning", "Fan Coil Unit", "VRV", "Heat Pump"},
        }
        return ad.data.heating_type in heat_mapping.get(parameter, {})
```